### PR TITLE
Feat/watchlater

### PR DESCRIPTION
**Pull request by abhij1607**
This PR consist of

- watchlater api's
- watchlater page
- add and delete video feature

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { Login } from "./pages/Login/Login";
 import { SignUp } from "./pages/Signup/Signup";
 import { Playlist } from "./pages/Playlist/Playlist";
 import { SinglePlaylist } from "./pages/SinglePlaylist/SinglePlaylist";
+import { WatchLater } from "./pages/WatchLater/WatchLater";
```
Comment (junaidshaikh-js) [inline src/App.js]: You can use barrel imports to avoid repetition

**Review by junaidshaikh-js — COMMENTED** [review]
Good work. Clean code. 

Suggested one change, everything else looks fine and can be merged.

File: src/pages/Home/HomeComponents/VideosPane/VideoCard/VideoCard.jsx [hunk 2]
```diff
@@ -4,26 +4,52 @@ import { useState } from "react";
 import { useAuth } from "../../../../../context/auth-context";
 import { useNavigate } from "react-router-dom";
 import { PlaylistModal } from "../../../../../components/PlaylistModal/PlaylistModal";
+import {
+  requestAddVideoInWatchLater,
+  requestDeleteVideoInWatchLater,
+} from "../../../../../utils/server-request";
```
Comment (Tanay-J) [inline src/pages/Home/HomeComponents/VideosPane/VideoCard/VideoCard.jsx:10]: You may create a jsconfig.json file in root folder and set base directory as /src
This way you can avoid writing .../... which would result in even cleaner code. You may consider this while finishing up the project.

ref: https://stackoverflow.com/questions/68675994/what-is-jsconfig-json check how to setup baseUrl in compilerOptions

File: src/components/Sidebar/Sidebar.jsx [hunk 3]
```diff
@@ -20,19 +20,19 @@ const Sidebar = () => {
...
-            <Link className="link" to="/">
+            <Link className="link" to="/watchlater">
               <i className="fa fa-bookmark" aria-hidden="true" />
               Watch Later
             </Link>
           </li>
           <li className="nav-list pd-y-base">
-            <Link className="link" to="/">
+            <Link className="link" to="/history">
               <i className="fa fa-history" aria-hidden="true" />
```
Comment (Tanay-J) [inline src/components/Sidebar/Sidebar.jsx:36]: Liked the use of accessibility attribute

**Review by Tanay-J — APPROVED** [review]
Good work at keeping the code clean and readable. Good to merge.

File: src/App.js (hunk 1 above)
Comment (abhij1607) [reply src/App.js]: thanks for the suggestion, will add barrel imports

File: src/pages/Home/HomeComponents/VideosPane/VideoCard/VideoCard.jsx (hunk 2 above)
Comment (abhij1607) [reply src/pages/Home/HomeComponents/VideosPane/VideoCard/VideoCard.jsx:10]: thanks for the suggestion tanay, will check more on this

**Comment by abhij1607** [comment]
used barrel imports as suggested by junaid

**Pull request merged**